Implemented cmap.

pcubillos opened the pull request:
Hi Francesco,
I was looking for a piece of code that transformed an RGBA colormap to an RGB colormap.  I couldn't find a solution, but I found your repo, and combined it with another piece of code to implement this new function.

pcubillos committed:
Implemented cmap.

montefra commented:
@pcubillos: thank you very much for the PR. It's very pleasant to know that someone uses this little project. I'll accept it for sure after checking the code.

montefra commented on mimic_alpha.py:
```diff
@@ -189,6 +191,81 @@ def colorAlpha_to_rgb(colors, alpha, bg='w'):
+       Value of alpha to mimic in range [0,1].
+    bg: Matplotlib color
+       Color of the background.
+
```
Add the return value, please:

```
Returns
----------
ma_cmap : :class:`matplotlib.colors.Colormap`
     add your explanation
```

montefra commented on mimic_alpha.py:
```diff
@@ -189,6 +191,81 @@ def colorAlpha_to_rgb(colors, alpha, bg='w'):
+        cdict['blue' ].append((pos, b, b))
+
+    # mimic-alpha colormap:
+    ma_cmap = mplc.LinearSegmentedColormap('my_colormap', cdict, 256)
```
I would make the the name of the color map an optional argument of the function.
Something like: 

```
def cmap(cmap_name, alpha, bg="w", out_cmap_name=None):
[...]
if not out_cmap_name:
    out_cmap_name = cmap_name + "_{0:.1f}".format(alpha)
ma_cmap = mplc.LinearSegmentedColormap(out_cmap_name, cdict, 256)
[...]
```

pcubillos commented:
Yes, I wasn't careful about that line with the output colormap name. I like your solution.
I'll update the changes and make a new pull request.
Thanks!

pcubillos committed:
Added optional argument to set the output colormap name.

pcubillos committed:
Added set_under, set_over, and set_bad optional arguments to the cmap
function.

pcubillos commented:
OK, I implemented the changes. I also implemented (and tested) the set_under, set_over, and set_bad optional arguments, and bumped the version up to 0.22

cheers,
Patricio.

montefra commented:
Thanks you very much for the addition. I'll merge a soon as I'm in front of a computer.
